### ISSUE TITLE
[BUG] Invalid tier value doesn't show valid options

### ISSUE THREAD

**Issue by chrismlittle123** [chrismlittle123/check-my-toolkit#161]
## Description

When an invalid tier value is used in `repo-metadata.yaml`, the validation silently falls back to "internal" without telling the user:
1. That their tier value is invalid
2. What the valid tier values are

## Steps to Reproduce

1. Create `check.toml`:
```toml
[extends]
registry = "github:example/standards"
rulesets = ["base-production"]
```

2. Create `repo-metadata.yaml` with invalid tier:
```yaml
tier: staging
```

3. Run:
```bash
cm validate tier
```

## Expected Result

Clear error about invalid tier value:
```
✗ Tier validation failed
  Error: Invalid tier 'staging'. Valid values are: production, internal, prototype
  
  Hint: Update repo-metadata.yaml to use a valid tier value.
```

## Actual Result

```
✗ Tier validation failed
  Tier: internal (source: default)
  Expected pattern: *-internal
  Rulesets: [base-production]
  Error: No ruleset matching pattern '*-internal' found. Rulesets: [base-production]
```

The user has no idea:
- That "staging" was rejected as invalid
- What valid tier values are
- Why `source: default` instead of `source: repo-metadata.yaml`

## Impact

Users will struggle to debug tier validation failures caused by typos or invalid values:
- `tier: staging` (invalid - not in valid list)
- `tier: prod` (invalid - must be `production`)
- `tier: dev` (invalid - must be `prototype`)
- `tier: Production` (invalid - case sensitive, covered by #147)

## Related Issues

- #147: Tier schema duplicated (includes case sensitivity)
- #156: Invalid YAML silently defaults
- #158: Empty file not distinguished from missing

All of these share the pattern of "silent fallback with no explanation".

## Suggested Fix

In `getTier()` function, when tier is not in VALID_TIERS:
```javascript
if (!VALID_TIERS.includes(tier)) {
    console.warn(`Warning: Invalid tier '${tier}'. Valid values are: ${VALID_TIERS.join(', ')}`);
    return { tier: DEFAULT_TIER, source: "default", invalidValue: tier };
}
```

## Environment

- check-my-toolkit version: 1.4.0
- Node.js version: 25.2.1
- Platform: macOS (Darwin 25.2.0)
